src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/AttackSystem.java

```diff
@@ -127,12 +84,11 @@ public void onInventorySlotChanged(InventorySlotChangedEvent event, EntityRef en
         if (itemIsFlag(item)) {
             String flagTeam = checkWhichFlagPicked(event);
             if (flagTeam.equals(player.getComponent(LASTeamComponent.class).team)) {
-                moveFlagToBase(player, flagTeam);
-                return;
+                player.send(new MoveFlagToBaseEvent(item, flagTeam));
```

Comment:
Why do we send this event against the player? I would have assumed that

- we send the event against the flag item (because that's what we want to move)
- `ReturnToBaseEvent` (yes, different name) would return the target to its base
- the target base is implicitly given by the team the flag belongs to